FV3/gfsphysics/GFS_layer/GFS_diagnostics.F90

```diff
@@ -229,13 +237,21 @@ subroutine GFS_externaldiag_populate (ExtDiag, Model, Statein, Stateout, Sfcprop
     ExtDiag(idx)%mod_name = 'gfs_phys'
     ExtDiag(idx)%cnvfac = cn_one
     ExtDiag(idx)%time_avg = .TRUE.
-    ExtDiag(idx)%time_avg_kind = 'rad_sw'
+    if (.not. Model%override_surface_radiative_fluxes) then
+      ExtDiag(idx)%time_avg_kind = 'rad_sw'
+    endif
     ExtDiag(idx)%intpl_method = 'bilinear'
     ExtDiag(idx)%coarse_graining_method = 'area_weighted'
     allocate (ExtDiag(idx)%data(nblks))
-    do nb = 1,nblks
-      ExtDiag(idx)%data(nb)%var2 => IntDiag(nb)%fluxr(:,3)
-    enddo
+    if (Model%override_surface_radiative_fluxes) then
+      do nb = 1,nblks
+        ExtDiag(idx)%data(nb)%var2 => IntDiag(nb)%uswsfc(:)
+      enddo
+    else
+      do nb = 1,nblks
+        ExtDiag(idx)%data(nb)%var2 => IntDiag(nb)%fluxr(:,3)
```

Comment:
what was the reason to use `uswsfc(:)` instead of `fluxr(:,3)` for the override case?

Reply:
Right now, the reason is that I am continuing to use `fluxr(:,3)` and `fluxr(:,4)` as the accumulation containers for the time-mean shortwave flux diagnostics from RRTMG.  It would make things more consistent with the other RRTMG diagnostics if I added a separate container for those, and continued to use `fluxr(:,3)` and `fluxr(:,4)` for the surface shortwave radiative fluxes "felt by the land surface model."  It would require some minor changes to the radiation driver, but I think that would be OK.

The conditional logic setting `ExtDiag(idx)%time_avg_kind` to `'rad_sw'` in the non-overriding case would still be required.